### ISSUE TITLE
Indexing gets very slow after ~8 380 000 entries

### ISSUE THREAD

**Issue by kgosselinCoe** [nextapps-de/flexsearch#140]
Basically i have 11 000 000+ companies to index, columns are 
- idCompany: int length up to 11
- name: str length up to 255
- identifyNumber: str length 9 [0-9]
- postalCode: str length 5

options: 
```
doc: {
  id: "idCompany",
  field:{
    "identifyNumber":{
      tokenize: "strict", 
      encode: "extra",
      depth: 0,
      threshold: 0,
      resolution: 1
    },
    "name":{
      tokenize: "reverse",
      encode: "advanced",
      threshold:2,
      resolution:9,
      depth:false
    },
    "postalCode":{
      tokenize: "strict",
      encode: "extra",
      depth: 0,
      threshold: 0,
      resolution: 1
    }
  }
}
```

I tried different methods to index my data: the whole thing, chunk by chunk(1M and 100k), and line by line.

My problem is that after 8 380 000 lines the indexing gets very slow, it goes from ~100k every 10 or 15 sec to 1 every 5 or 6 sec. 
I tried indexing only the idCompany and postalCode which should cut out most of the data and I only get to index ~8 500 000 and then it starts getting slow again.
Server has ~50G of free ram, the indexing has used ~12G for all the columns and ~4G when using only the idCompany and postalCode

Any idea why this happens ?

**Comment by ts-thomas** [comment]
Wow its a nice use case, I will made some load tests to figure this out. Probably it requires clustering. When 0.7.0 is finished the Worker interface gets support for Node.js

**Comment by KrishnaPG** [comment]
Came across this issue while trying to learn more about this package. Could be slightly off-topic, but I am interested in learning why not use a traditional database for 11M+ records data and instead use the FlexSearch?

Looks like FlexSearch holds all the data in-memory, which means if the process is terminated (for any reason) the whole index has to be created up and again upon restart (or is it possible to persist the structure to the disk and load to memory on start?).  Would it not make sense to use a persisted DB rather? Any benefit in using FlexSearch over traditional DBs? 

For example, a simple 8GB/16GB RAM machine with PostGres could handle the 11M+ records fine, I guess - correct?

**Comment by kgosselinCoe** [comment]
@KrishnaPG I don't know about PostGres (i use MySQL) 

I had a fulltext search engine (autocomplete like) working on MySQL but it lacked some features like reverse indexing or the tokenizer. 
I still use the fulltext search for some project, it can be very powerful when fine tuned but i needed something like flexsearch for this one.

In Flexsearch you can export/import the index and/or documents but i feel like it's not complete/optimized yet

For MySQL a 16GB database could handle 11M+ records depending on the requests you make and the number of request as it is not loaded in RAM like flexsearch. (for example we had a 30GB database but we switched to a 120GB recently because of user workload and our heavy ass requests combined to a fuckload of data)

**Comment by kgosselinCoe** [comment]
@ts-thomas just wanted to tell you i managed to do what i wanted to do.
Version used : 0.6.32

I've got ~12 million companies indexed, when garbage is cleared it uses around 15G of RAM

I used multiple indexes to split everything as exports won't work when there's too much data (JSON parse error) but i still got stuck with the slow indexing problem.
I got around that by indexing parts of it, moving files out of the project, restart indexing etc...
When i get all of my exports, i move everything back in the project, restart it and import everything in.

I have no idea on how or why this is working but... it works...  

When searching for a doc by it's ID  from looping all the indexes it's rather slow (~3 secs) but the search process is still fast ~50ms i haven't tried inserts/updates/deletes on the go yet
